go/sqlescape/ids.go

```diff
@@ -52,11 +52,21 @@ func EscapeIDs(identifiers []string) []string {
 	return result
 }
 
-// UnescapeID reverses any backticking in the input string.
+// UnescapeID reverses any backticking in the input string by EscapeID.
 func UnescapeID(in string) string {
```

Comment:
I do see your point, one concern is that returning an error could be a breaking change. Previously the function would handle cases like ``` `foo ```, `aa`, ``` a`a ``` etc by returning them as is.

If that's not an issue, I'll go ahead and make the changes

Reply:
I think we should make the change yeah. Because not doing it also carries a security risk since it means you can trust escaping anymore. That’s more serious imho.

Reply:
Added the suggested changes

Reply:
> Is this the right way to go about it? I'll have to refactor 27 usages of the `UnescapeID` function across 6 files.

Yep, that's the consequence here but I think that's good since those call sites now know if it couldn't actually be unescaped. And also further bubble up the error then.